src/bin/thermalprobe.rs

```diff
@@ -0,0 +1,51 @@
+#![no_std]
+#![no_main]
+#![feature(type_alias_impl_trait)]
+
+use defmt::*;
+use embassy_executor::Spawner;
+use embassy_stm32::gpio::{Level, Output, Speed};
+use embassy_time::Timer;
+use {defmt_rtt as _, panic_probe as _};
+use embassy_stm32::spi::{Config, Spi};
+use embassy_stm32::time::Hertz;
+use embassy_stm32::dma::NoDma;
+
+
+#[embassy_executor::main]
+async fn main(_spawner: Spawner) {
+    let p = embassy_stm32::init(Default::default());
+    info!("Hello World!");
+
+    let mut spi_config = Config::default();
+    //recommended speed
+    spi_config.frequency = Hertz(4_000_000);
+    //unclear, found resource suggesting mode_1 being necessary
+    spi_config.mode = embassy_stm32::spi::MODE_1;
+    //using SPI2 after SPI1 seemed to be non-functional. Need to test on additional boards to be sure
```

Comment:
Interesting, Will have to try this out with my bluepill as well.

Overall code looks good but I'm interested to know about potential SPI issues we might have.